core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -171,6 +176,28 @@ public void killAll(Process proc, Map<String, String> modelEnvVars) throws Inter
             }
         return killers;
     }
+    
+    final List<ProcessKillingVeto> getVetoers() throws InterruptedException {
```

Comment:
If I understand correctly, the class is recreated every time ProcessTree.get() is called, which appears to be at the end of any call to Launcher.  Does it matter if the new plugin isn't invoked for a currently running job? I think that's the worst case that would happen here (if even that much).